src/main/java/edu/harvard/iq/dataverse/api/util/JSONResponseBuilder.java

```diff
@@ -0,0 +1,267 @@
+package edu.harvard.iq.dataverse.api.util;
+
+import edu.harvard.iq.dataverse.api.ApiBlockingFilter;
+
+import javax.json.Json;
+import javax.json.JsonObjectBuilder;
+import javax.servlet.ServletResponse;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import java.io.IOException;
+import java.util.Optional;
+import java.util.UUID;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+public class JSONResponseBuilder {
```

Comment:
Since this is only for errors, could/should it be JSONErrorResponseBuilder?

Reply:
I would like to keep this open minded. For now its errors only, but it would be great not only to streamline error responses, but success stories, too. Like seen in `ok()`, ... at https://github.com/IQSS/dataverse/blob/352b17f09beac5e0a9f44eb8a06cae7b5f6d2905/src/main/java/edu/harvard/iq/dataverse/api/AbstractApiBean.java#L652-L747